src/models/linear_ticker_data.rs

```diff
@@ -3,125 +3,246 @@ use crate::prelude::*;
+            debug_struct.field("tick_direction", tick_direction);
+        }
+        if let Some(price_24h_pcnt) = &self.price_24h_pcnt {
+            debug_struct.field("price_24h_pcnt", price_24h_pcnt);
+        }
+        if let Some(last_price) = &self.last_price {
+            debug_struct.field("last_price", last_price);
+        }
+        if let Some(prev_price_24h) = &self.prev_price_24h {
+            debug_struct.field("prev_price_24h", prev_price_24h);
+        }
+        if let Some(high_price_24h) = &self.high_price_24h {
+            debug_struct.field("high_price_24h", high_price_24h);
+        }
+        if let Some(low_price_24h) = &self.low_price_24h {
+            debug_struct.field("low_price_24h", low_price_24h);
+        }
+        if let Some(prev_price_1h) = &self.prev_price_1h {
+            debug_struct.field("prev_price_1h", prev_price_1h);
+        }
+        if let Some(mark_price) = &self.mark_price {
+            debug_struct.field("mark_price", mark_price);
+        }
+        if let Some(open_interest) = &self.open_interest {
+            debug_struct.field("open_interest", open_interest);
+        }
+        if let Some(open_interest_value) = &self.open_interest_value {
+            debug_struct.field("open_interest_value", open_interest_value);
+        }
+        if let Some(turnover_24h) = &self.turnover_24h {
+            debug_struct.field("turnover_24h", turnover_24h);
+        }
+        if let Some(volume_24h) = &self.volume_24h {
+            debug_struct.field("volume_24h", volume_24h);
+        }
+        if let Some(next_funding_time) = &self.next_funding_time {
+            debug_struct.field("next_funding_time", next_funding_time);
+        }
+        if let Some(funding_rate) = &self.funding_rate {
+            debug_struct.field("funding_rate", funding_rate);
+        }
+        if let Some(bid_price) = &self.bid_price {
+            debug_struct.field("bid_price", bid_price);
+        }
+        if let Some(bid_size) = &self.bid_size {
+            debug_struct.field("bid_size", bid_size);
+        }
+        if let Some(ask_price) = &self.ask_price {
+            debug_struct.field("ask_price", ask_price);
+        }
+        if let Some(ask_size) = &self.ask_size {
+            debug_struct.field("ask_size", ask_size);
+        }
+        if let Some(pre_qty) = &self.pre_qty {
+            debug_struct.field("pre_qty", pre_qty);
+        }
+        if let Some(cur_pre_listing_phase) = &self.cur_pre_listing_phase {
+            debug_struct.field("cur_pre_listing_phase", cur_pre_listing_phase);
+        }
+        if let Some(pre_open_price) = &self.pre_open_price {
+            debug_struct.field("pre_open_price", pre_open_price);
+        }
+
+        debug_struct.finish()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use itertools::Itertools;
+
+    use crate::fixture;
+
+    use super::*;
+
+    #[test]
+    fn deserialize() {
+        let json = fixture!("ws_linear_ticker");
```

Comment:
thanks to the new fixture, which detects broken references **at compile time** (not runtime, huge benefit!) we can easily and **safely** open JSON files with fixtures (test data).